File: pythonExample/mock_api.py
import random
import time
from flask import Flask, jsonify

app = Flask(__name__)

@app.route('/pay', methods=['GET'])
def mock_payment_api():
    """Simulates an unreliable API with a 30% failure rate."""
    time.sleep(random.uniform(0.1, 0.5))  # Simulate network latency

    if random.random() < 0.3:  # 30% failure rate
        return jsonify({"error": "Payment API timeout!"}), 500

    return jsonify({"status": "success", "transaction_id": random.randint(1000, 9999)}), 200

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=5001)

File: jsExample/mockApi.js
const express = require('express');
const app = express();

app.get('/pay', (req, res) => {
    // Simulate network latency
    setTimeout(() => {
        // 40% failure rate
        if (Math.random() < 0.4) {
            return res.status(500).json({
                error: "Payment API timeout!"
            });
        }

        res.json({
            status: "success",
            transaction_id: Math.floor(Math.random() * 9000) + 1000
        });
    }, Math.random() * 400 + 100); // Random delay between 100-500ms
});

const PORT = 5005;
app.listen(PORT, '0.0.0.0', () => {
    console.log(`Mock API server running on port ${PORT}`);
}); 